xpk.py

```diff
@@ -6497,6 +6585,398 @@ def add_shared_arguments(custom_parser):
+    '--custom-tpu-nodepool-arguments',
+    type=str,
+    default='',
+    help=(
+        'DEPRECATING SOON! Please use --custom-nodepool-arguments to customize'
+        ' node pool create command. Do note, these will not override already'
+        ' used node pool creation arguments. Example usage'
+        ' --custom-tpu-nodepool-arguments="--enable-ip-alias"'
+    ),
+    )   
+
+
+def add_shared_cluster_create_tensorboard_arguments(create_args_parsers):
+  """Add shared tensorboard arguments in cluster create and Pathways cluster create.
+  Note that this feature enables non-Pathways workloads to use tensorboard arguments
+  on a Pathways cluster.
+  Args:
+      List of cluster create tensorboard arguments parsers
+  """
+  for custom_parser in create_args_parsers:
+    custom_parser.add_argument(
+        '--create-vertex-tensorboard',
+        action='store_true',
+        help='Set this flag to create a Tensorboard instance in Vertex AI.',
+    )
+    custom_parser.add_argument(
+        '--tensorboard-region',
+        type=str,
+        default='us-central1',
+        help=(
+            'The region to create Vertex Tensorboard instance in. Visit'
+            ' https://cloud.google.com/vertex-ai/docs/general/locations#available-regions'
+            ' to view regions supported by Vertex AI. By default, Tensorboard'
+            ' instance will be created in us-central1.'
+        ),
+    )
+    custom_parser.add_argument(
+        '--tensorboard-name',
+        type=str,
+        required=False,
+        help=(
+            'The name of Vertex Tensorboard instance to create. '
+            'If not specified, a Tensorboard instance with the name '
+            f'<cluster>-{_DEFAULT_VERTEX_TENSORBOARD_NAME} will be created.'
+        ),
+    )
+
+
+def add_shared_cluster_create_capacity_arguments(create_args_parsers):
+  """Add shared capacity arguments in cluster create and Pathways cluster create.
+
+  Args:
+      List of cluster create capacity arguments parsers
+  """
+  for custom_parser in create_args_parsers:
+    custom_parser.add_argument(
+        '--on-demand',
+        action='store_true',
+        help=(
+            'Sets node pool creation to use on-demand resources. '
+            ' See `--reservation` or `--spot` for other capacity types.'
+        ),
+    )
+    custom_parser.add_argument(
+        '--reservation',
+        type=str,
+        help=(
+            'The reservation to be used for acquiring resources in the'
+            ' cluster. This will attempt to find the provided reservation.'
+            ' See `--spot` or `--on-demand` for other capacity types.'
+        ),
+    )
+    custom_parser.add_argument(
+        '--spot',
+        action='store_true',
+        help=(
+            'Sets node pool creation to use spot resources.'
+            ' See `--reservation` or `--on-demand` for other capacity types.'
+        ),
+    )
+
+
+def add_shared_workload_create_required_arguments(create_args_parsers):
```

Comment:
nit workload_arg_parsers or just arg_parsers. I see some inconsistencies in the other workload shared argument functions.

Reply:
The intention here was to talk about the action "create" for both clusters and workloads. I will change it if that causes confusion, to keep something more generic.